core/src/test/java/org/elasticsearch/ElasticsearchExceptionTests.java

```diff
@@ -637,21 +659,148 @@ public void testFailureToAndFromXContent() throws IOException {
         }
         assertNotNull(parsedFailure);
 
-        ElasticsearchException expected = exceptions.v2();
-        if (detailed) {
-            assertDeepEquals(expected, parsedFailure);
+        String reason;
+        if (failure instanceof ElasticsearchException) {
+            reason = failure.getClass().getSimpleName() + "[" + failure.getMessage() + "]";
         } else {
-            String reason;
-            if (failure instanceof ElasticsearchException) {
-                reason = failure.getClass().getSimpleName() + "[" + failure.getMessage() + "]";
-            } else {
-                reason = "No ElasticsearchException found";
-            }
-            assertEquals(ElasticsearchException.buildMessage("exception", reason, null), parsedFailure.getMessage());
-            assertEquals(0, parsedFailure.getHeaders().size());
-            assertEquals(0, parsedFailure.getMetadata().size());
-            assertNull(parsedFailure.getCause());
+            reason = "No ElasticsearchException found";
         }
+        assertEquals(ElasticsearchException.buildMessage("exception", reason, null), parsedFailure.getMessage());
+        assertEquals(0, parsedFailure.getHeaders().size());
+        assertEquals(0, parsedFailure.getMetadata().size());
+        assertNull(parsedFailure.getCause());
+    }
+
+    public void testFailureToAndFromXContentWithDetails() throws IOException {
+        final XContent xContent = randomFrom(XContentType.values()).xContent();
+
+        Exception failure;
+        Throwable cause;
+        ElasticsearchException expected;
+        ElasticsearchException suppressed;
+
+        switch (randomIntBetween(0, 6)) {
+            case 0: // Simple elasticsearch exception without cause
+                failure = new NoNodeAvailableException("A");
```

Comment:
that's odd, the root cause is really useful only with SearchPhaseExecutionException. But I don't think we can do better than this ATM.